persist/fs/retriever.go

```diff
@@ -285,18 +321,39 @@ func (r *blockRetriever) Stream(
 	startTime time.Time,
 	onRetrieve block.OnRetrieveBlock,
 ) (xio.SegmentReader, error) {
-	reqs, err := r.shardRequests(shard)
-	if err != nil {
-		return nil, err
-	}
-
 	req := r.reqPool.Get()
 	req.shard = shard
 	req.id = id
 	req.start = startTime
 	req.onRetrieve = onRetrieve
 	req.resultWg.Add(1)
 
+	// This should never happen
+	if len(r.seekerMgrs) < 1 {
+		return nil, errNoSeekerMgrs
+	}
+	// It doesn't matter which seekerManager we use (they're all identical, we
+	// just have multiple for concurrency reasons), so we use the first one
+	// because it's guaranteed to be there (concurrency cannot be < 1)
+	seeker, err := r.seekerMgrs[0].Seeker(shard, startTime)
+	if err != nil {
+		return nil, err
+	}
+	// If the ID is not in the seeker's bloom filter, then it's definitely not on
+	// disk and we can return immediately
+	if !seeker.ConcurrentIDBloomFilter().Test(id.Data().Get()) {
```

Comment:
Sorry to flip flop on this, I thought about this again and it definitely would cause a lot of extra concurrency and scheduling for not much gain to keep the contract the same for both bloom filter lookups that are negative and actual lookups that are negative after searching disk.

I think the issue before was that we were returning `errSeekIDNotFound` which is impossible to test for equality as its not an exported error from the `fs` package.  We should either update the interface to be xplicilt

Reply:
After syncing looks like we can do this without any goroutine scheduling as we will no longer launch a goroutine for onRetrieve if there is no data.

Reply:
Spoke in person, decided that the existing implementation is fast enough, but that we should not call `go req.onRetrieve.OnRetrieveBlock()` if there is no data (Not found in files or not in bloom filter)